geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStore.scala

```diff
@@ -162,34 +163,13 @@ abstract class GeoMesaDataStore[DS <: GeoMesaDataStore[DS]](val config: GeoMesaD
 
   @throws(classOf[IllegalArgumentException])
   override protected def preSchemaUpdate(sft: SimpleFeatureType, previous: SimpleFeatureType): Unit = {
-    // check for attributes flagged 'index' and convert them to sft-level user data
-    sft.getAttributeDescriptors.asScala.foreach { d =>
-      val index = d.getUserData.remove(AttributeOptions.OptIndex).asInstanceOf[String]
-      if (index == null || index.equalsIgnoreCase(IndexCoverage.NONE.toString) || index.equalsIgnoreCase("false")) {
-        // no-op
-      } else if (index.equalsIgnoreCase(IndexCoverage.FULL.toString) || java.lang.Boolean.valueOf(index)) {
-        val fields = Seq(d.getLocalName) ++ Option(sft.getGeomField) ++ sft.getDtgField
-        val attribute = IndexId(AttributeIndex.name, AttributeIndex.version, fields, IndexMode.ReadWrite)
-        val existing = sft.getIndices.map(GeoMesaFeatureIndex.identifier)
-        if (!existing.contains(GeoMesaFeatureIndex.identifier(attribute))) {
-          sft.setIndices(sft.getIndices :+ attribute)
-        }
-      } else {
-        throw new IllegalArgumentException(s"Configured index coverage '$index' is not valid: expected " +
-            IndexCoverage.FULL.toString)
-      }
-    }
-
+    updateSchemaUserData(sft, previous)
     // try to create the new indices to ensure they are valid for the sft
-    val previousIndices = previous.getIndices.map(GeoMesaFeatureIndex.identifier)
-    val newIndices = sft.getIndices.filterNot(i => previousIndices.contains(GeoMesaFeatureIndex.identifier(i)))
-    if (newIndices.nonEmpty) {
-      try { GeoMesaFeatureIndexFactory.create(this, sft, newIndices) } catch {
-        case NonFatal(e) => throw new IllegalArgumentException(s"Error configuring new feature index:", e)
-      }
+    try { GeoMesaFeatureIndexFactory.create(this, sft, sft.getIndices) } catch {
```

Comment:
This seems like it'll call create on existing indexes after this change.  Is that operation a no-op if the tables exist?

Reply:
this just creates the configured classes in memory (i.e. Z3Index), the actual tables are created later on here: https://github.com/locationtech/geomesa/pull/2615/files/e031f94d0c1cea62f8970d53c0a8e0f5d1a0fb35#diff-06656c6f4398571e1b268f89abc06664R223
(although calling create table is supposed to be coded as a no-op if the table already exists).

Reply:
this call ensures that they are SPI loadable.